bindings/jupyter-modules/jupyter-ma-causal/src/reducer.js

```diff
@@ -0,0 +1,54 @@
+import {handleActions} from 'redux-actions';
+import {scaleLinear} from 'd3-scale';
+import {extent as d3Extent} from 'd3-array';
+
+import {UPDATE_DATA, UPDATE_SLIDER_VALUES} from './actions';
+
+const DEFAULT_STATE = {
+  containerWidth: 800,
+  containerHeight: 800,
+  data: [],
+  sliderValues: [], // associative array
+};
+
+const handleUpdateData = (state, {payload}) => {
+  // temporary hack to scale the data until the actual scale method is implemented
+  payload.forEach(d => {
+    d.lines.forEach(({name, line}) => {
+      if (name === 'ipvu') {
+        const extent = d3Extent(line, d => d.y);
+        const scale = scaleLinear()
+          .domain(extent)
+          .range([0, 1]);
+        line.forEach(d => {
+          d.y = scale(d.y);
+        });
+      }
+    });
+  });
```

Comment:
Any reason why are modifying `payload` directly without creating a new array/object?

Reply:
This is only a temporary hack. We expect the input data to be scaled to [0, 1] from the Python code, because of this we can just modify the payload to make it match what we expect for now.